model/fn-execution/src/main/proto/org/apache/beam/model/fn_execution/v1/beam_fn_api.proto

```diff
@@ -137,12 +138,79 @@ message InstructionResponse {
     FinalizeBundleResponse finalize_bundle = 1004;
     MonitoringInfosMetadataResponse monitoring_infos = 1005;
     HarnessMonitoringInfosResponse harness_monitoring_infos = 1006;
+    SampleDataResponse sample = 1007;
 
     // DEPRECATED
     RegisterResponse register = 1000;
   }
 }
 
+// If supported, the `SampleDataRequest` will respond with a
+// `SampleDataResponse`. The SDK being queried must have the
+// "beam:protocol:data_sampling:v1" capability. The samples in the
+// `SampleDataResponse` will be filtered by both being in the
+// `process_bundle_descriptor_ids` AND the `pcollection_ids`. An empty list can
```

Comment:
Are you use you want AND here vs OR?

What kinds of lookups do you think runners will ask for?

Reply:
Clarified in the comment that samples are returned based on PBD ids then filtered on PCollection ids. In this way, the runner can ask for:

- All samples
- All Samples from a particular PBD
- All samples from a particular PCollection (regardless of PBD)
- Samples from a set of PCollections in a PBD

Reply:
That sounds like you want to use an `OR` clause since `All samples from a particular PCollection (regardless of PBD)` and `Samples from a set of PCollections in a PBD` are the same when the PCollection ids are unique.

Reply:
Sounds good! Used your suggested commit